backend/src/main/java/com/karankumar/bookproject/backend/model/Book.java

```diff
@@ -244,6 +246,20 @@ public BookBuilder edition(Integer edition) {
         }
     }
 
+    public void addPublisher(@NonNull Publisher publisher) {
+        Set<Publisher> bookPublishers = this.getPublishers();
+        if (bookPublishers == null) {
```

Comment:
Instead of this null check, could you initialise the set to an empty hash set where we're declaring the field?